src/frontend/App.tsx

```diff
@@ -21,29 +23,35 @@ const App = () => {
       <Button
         title="Send message"
         disabled={messageText.length === 0}
-        onPress={() => channel.send(messageText)}
+        onPress={async () => {
+          try {
+            const res = await clientApi?.greet(messageText);
+            console.log('rpc call', res);
+          } catch (e) {
+            console.log('error sendind rpc', e);
+          }
+          try {
+            const res = await clientApi?.asyncGreet(messageText);
+            console.log('async rpc call', res);
+          } catch (e) {
+            console.log('error sendind async rpc', e);
+          }
+        }}
       />
     </SafeAreaView>
   );
 };
 
 function useNodejsMobile() {
+  const [clientApi, setClientApi] = React.useState<typeof MapeoClient>();
   React.useEffect(() => {
     nodejs.start('loader.js');
-
-    const subscription = nodejs.channel.addListener('message', msg => {
-      console.log('RECEIVED MESSAGE', msg);
-    });
-
+    const channel = new MessagePortLike(nodejs.channel);
+    setClientApi(createClient<typeof MapeoClient>(channel));
     return () => {
-      // @ts-expect-error
-      subscription.remove();
+      channel.close();
     };
   }, []);
-
-  return {
-    send: (msg: string) => nodejs.channel.send(msg),
-  };
+  return clientApi;
```

Comment:
`nodejs.start()` shouldn't be in here - that should just happen once when the JS file first loads, not tied to the react lifecycle.

I'm not sure if the channel and clientApi need to be in here either, nor tied to the useEffect. This isn't a side-effect, and there is no async here, so this code is rendering the app once with no clientApi, then calling `setClientApi()` and rendering again. If we want to do a `useClientApi()` it should probably just be a wrapper for `useMemo()`, since we only want it once, or we could just put the client api in the global scope.

Reply:
ah yeah this stuff I worked on - not sure why I did this but fully agree that it should be set up outside of react

Reply:
ah remember why i had it set up this way - mapeo mobile does it similarly (where `api.startServer` calls `nodejs.start()`): 

https://github.com/digidem/mapeo-mobile/blob/ca167c257b90ba74b0e63e623f405aed98d86145/src/frontend/AppLoading.tsx#L77-L81

either way, agree it probably could just be moved outside for the sake of this project

Reply:
addressed in https://github.com/digidem/react-native-with-mapeo-core-next/pull/7